daml-assistant/src/DAML/Assistant/Env.hs

```diff
@@ -59,11 +59,51 @@ getDamlEnv = do
     envLatestStableSdkVersion <- getLatestStableSdkVersion envDamlPath
     pure Env {..}
 
+-- | (internal) Override function with environment variable
+-- if it is available.
+overrideWithEnvVar
+    :: String                   -- ^ env var name
+    -> (String -> t)            -- ^ parser for env var
+    -> IO t                     -- ^ calculation to override
+    -> IO t
+overrideWithEnvVar envVar parse calculate = do
+    valueM <- getEnv envVar
+    case valueM of
+        Nothing -> calculate
+        Just value -> pure (parse value)
+
+-- | (internal) Same as overrideWithEnvVar but accepts "" as
+-- Nothing and throws exception on parse failure.
+overrideWithEnvVarMaybe
+    :: Exception e
+    => String                   -- ^ env var name
+    -> (String -> Either e t)   -- ^ parser for env var
+    -> IO (Maybe t)             -- ^ calculation to override
+    -> IO (Maybe t)
+overrideWithEnvVarMaybe envVar parse calculate = do
```

Comment:
I feel like we should be able to simplify this somehow but I also don’t have a good suggestion.